language-support/ts/codegen/tests/npm_check.sh

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env bash
+# Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# This test checks that the package-lock.json file gets updated by npm when a generated dependency
+# changes.
+
+set -eou pipefail
+
+# --- begin runfiles.bash initialization v2 ---
+# Copy-pasted from the Bazel Bash runfiles library v2.
+    set -uo pipefail; f=bazel_tools/tools/bash/runfiles/runfiles.bash
+    source "${RUNFILES_DIR:-/dev/null}/$f" 2>/dev/null || \
+      source "$(grep -sm1 "^$f " "${RUNFILES_MANIFEST_FILE:-/dev/null}" | cut -f2- -d' ')" 2>/dev/null || \
+      source "$0.runfiles/$f" 2>/dev/null || \
+      source "$(grep -sm1 "^$f " "$0.runfiles_manifest" | cut -f2- -d' ')" 2>/dev/null || \
+      source "$(grep -sm1 "^$f " "$0.exe.runfiles_manifest" | cut -f2- -d' ')" 2>/dev/null || \
+      { echo>&2 "ERROR: cannot find $f"; exit 1; }; f=; set -e
+# --- end runfiles.bash initialization v2 ---
+
+DAML=$(rlocation "$TEST_WORKSPACE/$1")
+NPM=$(rlocation "$TEST_WORKSPACE/$2")
+SDK_VERSION=$3
+TMP_DIR=$(mktemp -d)
+
+cleanup() {
+    rm -rf "$TMP_DIR"
+}
+trap cleanup EXIT
+
+mkdir -p "$TMP_DIR"/daml
+
+cd "$TMP_DIR"
+cat <<EOF > "$TMP_DIR"/daml.yaml
+sdk-version: $SDK_VERSION
+name: npmcheck
+source: daml
+version: 0.0.1
+dependencies:
+  - daml-stdlib
+  - daml-prim
+EOF
+
+cat <<EOF > "$TMP_DIR"/daml/Main.daml
+module Main where
+
+template T
+  with
+    p : Party
+  where
+    signatory p
+    choice C : Int
+      with
+        controller p
+          do return 42
+EOF
+
+cat <<EOF > "$TMP_DIR"/package.json
+{
+  "name": "test",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@daml.js/npmcheck": "file:daml.js/npmcheck-0.0.1"
+  }
+}
+EOF
+
+$DAML build -o npmcheck.dar
+$DAML codegen js npmcheck.dar -o "$TMP_DIR"/daml.js
+$NPM install
+PACKAGELOCK1=$(md5sum "$TMP_DIR"/package-lock.json | awk '{print $1}')
+
+rm "$TMP_DIR"/package-lock.json
+rm "$TMP_DIR"/daml/Main.daml
+
+cat <<EOF > "$TMP_DIR"/daml/Main.daml
+module Main where
+
+type X = (Party, Party)
+
+template T
+  with
+    p : Party
+  where
+    signatory p
+    choice C : X
+      with
+        controller p
+          do return (p, p)
+EOF
+
+rm npmcheck.dar
+$DAML build -o npmcheck.dar
+$DAML codegen js npmcheck.dar -o "$TMP_DIR"/daml.js
+$NPM install
+PACKAGELOCK2=$(md5sum "$TMP_DIR"/package-lock.json | awk '{print $1}')
```

Comment:
Can we throw `jq` at this to get the actual dependencies and verify that the list got longer by 1?

Reply:
using jq now.